squad/api/rest.py

```diff
@@ -503,53 +550,66 @@ def email(self, request, pk=None):
                 pass
 
         baseline = None
+        delayed_report, created = self.get_object().delayed_reports.get_or_create(
+            baseline=baseline,
+            template=template,
+            output_format=output_format,
+            email_recipient=email_recipient,
+            callback=callback,
+            callback_token=callback_token,
+            data_retention_days=data_retention_days
+        )
         if baseline_id is not None:
             try:
                 previous_build = Build.objects.get(pk=baseline_id)
                 baseline = previous_build.status
             except Build.DoesNotExist:
                 data = {
-                    "message": "Build %s does not exist" % baseline_id
+                    "message": "Baseline build %s does not exist" % baseline_id
                 }
-                return Response(data, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+                # return created=False to avoid running prepare_report
+                return save_delayed_report(delayed_report, data, status.HTTP_500_INTERNAL_SERVER_ERROR), False
```

Comment:
in squad/ci/tasks.py you used the following snippet before `save_delayed_report`:
```
if delayed_report.template is not None:
            data.update({
                "txt": delayed_report.template.plain_text,
                "html": delayed_report.template.html
            })
```

is the intention here not to have those extra args for `data`?

Reply:
In this line text and html part are not filled in yet. Since baseline doesn't exist it's not possible to create that content. But you're right it's a bit inconsistent.